### PR TITLE
Change algorithm for grabbing plaint-text file

### PR DESCRIPTION
**Pull request by DmitryAstafyev**

### CODE REVIEW

File: application/apps/indexer/processor/src/grabber.rs [hunk 1]
```diff
@@ -620,3 +591,19 @@ fn subtract_what_is_possible(n: u64, m: u64) -> u64 {
...
     }
 }
+
+/*
+    Should returns positions of all bytes, where new entity ends. It's called
+    "inlet" - point of entry's end.
+    TODO: probably it should be switched to BEGIN, but not END
+*/
+fn get_inlets_plaint_text(buffer: &[u8]) -> Result<(u64, usize), GrabError> {
+    let nl_count = bytecount::count(&buffer, b'\n') as u64;
```
Comment (marcmo) [inline application/apps/indexer/processor/src/grabber.rs]: do we need this call at all? after all, it should also work without that and I guess it is not really an optimization to do this call

File: application/apps/indexer/processor/src/grabber.rs [hunk 2]
```diff
@@ -620,3 +591,19 @@ fn subtract_what_is_possible(n: u64, m: u64) -> u64 {
...
+    if nl_count == 0 {
+        Ok((0, 0))
+    } else if let Some(offset) = buffer.iter().rposition(|&byte| (&byte == &b'\n' || &byte == &b'\r')) {
```
Comment (marcmo) [inline application/apps/indexer/processor/src/grabber.rs]: we should either check for `\n` in all places or `\n` and `\r`

File: application/apps/indexer/processor/src/grabber.rs (hunk 1 above)
Comment (DmitryAstafyev) [reply application/apps/indexer/processor/src/grabber.rs]: Do you mean `nl_count`? It's needed to create a slot  as soon as we have to understand how many rows we have inside of bytes chunk. 
Or your are talking about function `get_inlets_plaint_text`? This function I think should become a part of `trait` to make creating of map more general.

File: application/apps/indexer/processor/src/grabber.rs (hunk 2 above)
Comment (DmitryAstafyev) [reply application/apps/indexer/processor/src/grabber.rs]: Probably right way would be:
- check for `\n\r` or `\r\n` and use is as a single line-breaker -> exit
- check for `\n` -> exit
- check for `\r` -> exit
What do you think?

**Pull request merged**